src/dto/targets-index.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { IsInt, IsNotEmpty, Min, Max } from 'class-validator'
+
+import { Topic } from '../topics/topic.entity'
+import { TARGET_MAX_RADIUS, TARGET_MIN_RADIUS } from '../constants'
+
+export default class TargetIndexDto {
+  constructor(target) {
```

Comment:
Doesn't nest automatically transform these? You should have a data mapper and not do it by hand